recipes/automake/all/conanfile.py

```diff
@@ -1,140 +1,181 @@
+from conan.tools.scm import Version
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.51.3"
 
 
 class AutomakeConan(ConanFile):
     name = "automake"
+    description = "Automake is a tool for automatically generating Makefile.in files compliant with the GNU Coding Standards."
+    license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.gnu.org/software/automake/"
-    description = "Automake is a tool for automatically generating Makefile.in files compliant with the GNU Coding Standards."
     topics = ("conan", "automake", "configure", "build")
-    license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
     settings = "os", "arch", "compiler", "build_type"
 
-    exports_sources = "patches/*"
-
-    _autotools = None
-
     @property
-    def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
+    def _settings_build(self):
+        # TODO: Remove for Conan v2
+        return getattr(self, "settings_build", self.settings)
 
     @property
-    def _version_major_minor(self):
-        [major, minor, _] = self.version.split(".", 2)
-        return '{}.{}'.format(major, minor)
+    def win_bash(self):
+        return self._settings_build.os == "Windows"
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
 
     def configure(self):
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
+        try:
+            del self.settings.compiler.libcxx  # for plain C projects only
+        except ValueError:
+            pass
+        try:
+            del self.settings.compiler.cppstd  # for plain C projects only
+        except ValueError:
+            pass
+
+    def layout(self):
+        basic_layout(self, src_folder="automake")
 
     def requirements(self):
         self.requires("autoconf/2.71")
-        # automake requires perl-Thread-Queue package
-
-    def build_requirements(self):
-        if hasattr(self, "settings_build"):
-            self.build_requires("autoconf/2.71")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        self.requires("m4/1.4.19")
 
     def package_id(self):
-        del self.info.settings.arch
-        del self.info.settings.compiler
-        del self.info.settings.build_type
+        try:
+            del self.info.settings.arch
+        except ValueError:
+            pass
+        try:
+            del self.info.settings.compiler
+        except ValueError:
+            pass
+        try:
+            del self.info.settings.build_type
+        except ValueError:
+            pass
```

Comment:
The exception can only be triggered when `self.info.settings` is not set right? why not write
```suggestion
        if self.info.settings:
            del self.info.settings.arch
            del self.info.settings.compiler
            del self.info.settings.build_type
```

Reply:
https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#settings